### PR TITLE
Tseys/astree

### PR DESCRIPTION
**Pull request by SeysT**
This PR implements parsing code to ASTree and converting ASTree to CoverGraph.

Ready to be reviewed.

### CODE REVIEW

File: Examples/exemple.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+if 1 : (X <= 0)
+	then 2 : X := (0 - X)
+	else 3 : X := (1 - X);
+if 4 : (X=1)
+	then 5 :X:=1
+	else 6 :X :=(X+1)
```
Comment (yoagauthier) [inline Examples/exemple.txt:6]: Just need to add `0 : X := some_number;`at the beginning to allow the eval of the program, but I will do it in another PR

File: model/cover_graph.py [hunk 2]
```diff
@@ -1,26 +1,55 @@
...
+                if edge.root_vertex == old_vertex
+                else edge.root_vertex
+            )
+            edge.child_vertex = (
+                new_vertex
+                if edge.child_vertex == old_vertex
+                else edge.child_vertex
+            )
+            new_edges.append(edge)
+        return new_edges
```
Comment (yoagauthier) [inline model/cover_graph.py:25]: I do not really understand what this method does ... I understood it's needed when we have a sequence of instructions, like to merge the existing edges, but does that create new edges ? If so what for ?

**Review by yoagauthier — APPROVED** [review]
I've not read all the parsing code, but from some tests the output seems ok to me.

File: model/cover_graph.py (hunk 2 above)
Comment (SeysT) [reply model/cover_graph.py:25]: Ok I'll try to explain this with the example of the creation of the grpah for the sequence node. You have 2 different graphs with one input and one output for both of them. The idea of the new graph for the sequence node is to take the output vertex of the first graph and replace it by the input of the second one. To achieve this you want to replace the first output by the second input in the edges list of the first graph. This method returns this new edges list in order to not change the original edges list. This idea is to replace the old vertex by the new vertex in all edges.

Comment (yoagauthier) [reply model/cover_graph.py:25]: Ok got it, quite clever 👍

Comment (yoagauthier) [reply model/cover_graph.py:25]: Ok got it, quite clever 👍

**Pull request merged**